cmd/clusterctl/client/repository/repository_versions.go

```diff
@@ -39,9 +39,14 @@ func latestContractRelease(repo Repository, contract string) (string, error) {
 	}
 	// Attempt to check if the latest release satisfies the API Contract
 	// This is a best-effort attempt to find the latest release for an older API contract if it's not the latest release.
-	// If an error occurs, we just return the latest release.
 	file, err := repo.GetFile(latest, metadataFile)
+	// If an error occurs, we just return the latest release.
 	if err != nil {
+		if errors.Is(err, ErrNotFound) {
+			// If it was ErrNotFound, then there is no release yet for the resolved tag.
+			// Ref: https://github.com/kubernetes-sigs/cluster-api/issues/7889
+			return "", errors.Wrap(err, "release does not exist yet, consider setting \"GOPROXY=off\" as workaround")
```

Comment:
I'm pretty sure there are better wordings than this, happy for other suggestions :-)